app/upgrades/v20/upgrades.go

```diff
@@ -0,0 +1,144 @@
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/evmos/evmos/v19/contracts"
+	evmostypes "github.com/evmos/evmos/v19/types"
+	erc20keeper "github.com/evmos/evmos/v19/x/erc20/keeper"
+	erc20types "github.com/evmos/evmos/v19/x/erc20/types"
+	evmoscore "github.com/evmos/evmos/v19/x/evm/core/core"
+	"github.com/evmos/evmos/v19/x/evm/core/vm"
+	evmkeeper "github.com/evmos/evmos/v19/x/evm/keeper"
+	"github.com/evmos/evmos/v19/x/evm/statedb"
+)
+
+// CreateUpgradeHandler creates an SDK upgrade handler for v19
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	bk bankkeeper.Keeper,
+	erc20k erc20keeper.Keeper,
+	ek *evmkeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		logger := ctx.Logger().With("upgrade", UpgradeName)
+		logger.Info("Running migrations...")
+		if err := AddCodeToERC20Extensions(ctx, erc20k, ek, bk); err == nil {
+			return nil, err
+		}
+
+		return mm.RunMigrations(ctx, configurator, vm)
+	}
+}
+
+// AddCodeToERC20Extensions adds code and code hash to the ERC20 precompiles with the EVM.
+func AddCodeToERC20Extensions(ctx sdk.Context,
+	erc20Keeper erc20keeper.Keeper,
+	evmKeeper *evmkeeper.Keeper,
+	bankKeeper bankkeeper.Keeper,
+) error {
+	// need the interpreter to get the code after running the constructor
+	interpreter, err := getInterpreter(ctx, evmKeeper)
+	if err != nil {
+		return err
+	}
+
+	erc20Keeper.IterateTokenPairs(ctx, func(tokenPair erc20types.TokenPair) bool {
+		meta, found := bankKeeper.GetDenomMetaData(ctx, tokenPair.Denom)
+		if !found {
+			err = fmt.Errorf("no bank metadata found for %s", tokenPair.Denom)
+			return true
+		}
+
+		decimals := uint8(0)
+		if len(meta.DenomUnits) > 0 {
+			decimalsIdx := len(meta.DenomUnits) - 1
+			decimals = uint8(meta.DenomUnits[decimalsIdx].Exponent)
+		}
+
+		var ctorArgs []byte
+		ctorArgs, err = contracts.ERC20MinterBurnerDecimalsContract.ABI.Pack(
+			"",
+			meta.Name,
+			meta.Symbol,
+			decimals,
+		)
+		if err != nil {
+			return true
+		}
+
+		data := make([]byte, len(contracts.ERC20MinterBurnerDecimalsContract.Bin)+len(ctorArgs))
+		copy(data[:len(contracts.ERC20MinterBurnerDecimalsContract.Bin)], contracts.ERC20MinterBurnerDecimalsContract.Bin)
+		copy(data[len(contracts.ERC20MinterBurnerDecimalsContract.Bin):], ctorArgs)
+
+		contractAddr := common.HexToAddress(tokenPair.Erc20Address)
+		contract := vm.NewContract(vm.AccountRef(erc20types.ModuleAddress), vm.AccountRef(contractAddr), common.Big0, 200000)
+		contract.Code = data
+		contract.CodeHash = crypto.Keccak256Hash(data)
+
+		ret, err := interpreter.Run(contract, nil, false)
+		if err != nil {
+			return true
+		}
+
+		codeHash := crypto.Keccak256(ret)
+		evmKeeper.SetCode(ctx, codeHash, ret)
```

Comment:
are we setting the code too?

Reply:
I think will be better for EVM compatibility. Not having the code may break some EVM tooling, apps